### ISSUE TITLE
Improving Keyboard Navigation in Rancher Desktop

### ISSUE THREAD

**Issue by rak-phillip** [rancher-sandbox/rancher-desktop#4901]
Keyboard navigation in Rancher Desktop deviates from standard conventions promoted by each supported operating system (macOS, Windows, Linux).

Rancher Desktop currently implements the following keyboard shortcuts for menu navigation:

- Promote Navigation: `CommandOrControl + ]`
- Previous Navigation: `CommandOrControl + [`
- Navigate to specific menu item: `CommandOrControl + { num }`

These shortcuts were devised to mirror controls in applications like Numbers or Jetbrains IDEs. However, it would make sense to augment these existing shortcuts by adding keyboard navigation that is promoted by our supported platforms.

## Proposed Changes

Align Rancher Desktop's keyboard navigation with platform standards to provide a more consistent and intuitive experience for macOS, Windows, and Linux users.

### Most common rules shared across all platforms

- Move to the next UI element: Press the Tab key.
- Move to the previous UI element: Press Shift-Tab.
- Select an item: Press the Space bar.
- Move inside a group (such as a list of files in a folder or items in a sidebar): Press the Up Arrow or Down Arrow key, or the Left Arrow or Right Arrow key. Full Keyboard Access highlights the group and the item within the group that has focus.

### macOS

Refer to the [macOS Keyboard Navigation Documentation](https://support.apple.com/guide/mac-help/navigate-your-mac-using-full-keyboard-access-mchlc06d1059/13.0/mac/13.0) for the standard keyboard navigation guidelines on macOS.

### Windows

Refer to the [Windows Keyboard Shortcuts Documentation](https://support.microsoft.com/en-us/windows/keyboard-shortcuts-in-apps-139014e7-177b-d1f3-eb2e-7298b2599a34) for the standard keyboard navigation guidelines on Windows.

#### Gnome

Refer to the [Gnome Keyboard Navigation Documentation](https://help.gnome.org/users/gnome-help/stable/keyboard-nav.html.en) for the standard keyboard navigation guidelines on Gnome.

#### KDE

KDE doesn't define strong guidelines for this behavior. Implementing the most common rules mentioned above should be appropriate. For reference, you can check the [KDE Accessibility Checklist](https://develop.kde.org/hig/accessibility/checklist/).
